charts/spire/charts/spire-server/templates/issuer.yaml

```diff
@@ -0,0 +1,47 @@
+{{- with .Values.upstreamAuthority.certManager }}
+{{ if and .enabled .createCA }}
+{{/*
+Configuring CA Issuer: https://cert-manager.io/docs/configuration/ca/
+*/}}
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: {{ include "spire-server.fullname" $ }}-signed
+  namespace: {{ include "spire-server.namespace" $ }}
+  labels:
+    {{- include "spire-server.labels" $ | nindent 4}}
+spec:
+{{- if .signedSpec }}
+  {{- toYaml .signedSpec | nindent 4 }}
```

Comment:
Been trying to figure out some better place to point users for documentation on this, and not finding much. This section gets plugin specific pretty fast... But that raises another question. Do we really believe a user will use anything other then self signed here to get a CA level certificate, and at the same time, want to manage it via the chart and not themselves? With other plugins, its likely they will have to upload some other things such as credentials to talk to other systems too?

Reply:
aws private certificate authority would require indeed some annotation on cert-manager to allow for that usecase. That annotation is already supported by cert-manager. for others I can't judge.

Reply:
There a reason to use the aws private cert plugin to cert-manager over https://github.com/spiffe/spire/blob/main/doc/plugin_server_upstreamauthority_aws_pca.md ?

Reply:
Depends on personal preference. You could do both. But sometimes the people that need to operate the cluster prefer to do all the certificate integration into the cluster in a single place e.g. cert manager. and handle within the cluster from cert-manager.